src/repr/src/scalar.rs

```diff
@@ -558,6 +579,11 @@ impl fmt::Display for Datum<'_> {
                 f.write_str("\"")
             }
             Datum::Uuid(u) => write!(f, "{}", u),
+            Datum::Array(array) => {
+                f.write_str("[")?;
```

Comment:
Postgres docs say the text form of arrays uses curly braces, but I'm not sure what the guarantees of Display are supposed to be.

Reply:
Ah, good point. No guarantees on the `Display` impl whatsoever, but doesn't hurt to make it less confusing. Updated.